src/server/migrations/20200317230828_add_user_roles_schema.js

```diff
@@ -0,0 +1,37 @@
+exports.up = function(knex) {
+  return knex.schema.createTable('user_roles', function(table) {
+    table
+      .increments('id')
+      .notNullable()
+      .primary();
+    table
+      .integer('fk_role_id')
+      .unsigned()
+      .notNullable();
+    table
+      .integer('fk_user_id')
+      .unsigned()
+      .notNullable();
+    table
+      .timestamp('updated_at')
+      .defaultTo(knex.fn.now())
+      .notNullable();
+    table
+      .timestamp('created_at')
+      .defaultTo(knex.fn.now())
+      .notNullable();
+    table.timestamp('deleted_at');
+    table
+      .foreign('fk_role_id', 'fk_role_id')
```

Comment:
if we do not specify the foreign key name then the name is created like this:
user_roles_fk_role_id_foreign
I wanted to have name fk_role_id_foreign, that's why it is twice
but we can change it later based on requirements

Reply:
That looks fine, the example on knexjs documentation has a slightly different syntax that is easier to read because it doesn't repeat the name of the field (.foreign('fk_role_id', 'fk_role_id') but if you tested your syntax and it works go ahead. If you have any problem with it use this example http://knexjs.org/#Schema-foreign